src/controllers/userController.ts

```diff
@@ -0,0 +1,180 @@
+                    BaseController.showMessage('Данные изменены');
+                    // const router = new Router("#root");
+                    // router.go("/settings");
+                }
+            )
+            .catch(BaseController.showMessage);
+    }
+
+
+    async uploadAvatar(model: FormData) {
+        console.log('UserController.uploadAvatar()')
+        await UserApi.uploadAvatar(model)
+            .then(
+                (data: IUserInfo) => {
+                    console.log('success uploadAvatar');
+                    console.log('IUserInfo:', data);
+                    BaseController.showMessage('Аватар изменен');
+                    const router = new Router("#root");
+                    router.go("/settings");
+                }
+            )
+            .catch(BaseController.showMessage);
+    }
+
+
+    async processChangePasswordSubmit(e: SubmitEvent) {
+        e.preventDefault();
+        console.log('submit: UserController processChangePasswordSubmit');
+        const resObj = validationTypedSubmitHandler<ChangePasswordViewFormModel>(e, ChangePasswordViewFormModel);
+        console.log(resObj);
+        const canSendDataToLogin = resObj.isValidated;
+        if (canSendDataToLogin) {
+            if (resObj.object.newPassword != resObj.object.password_repeat) {
+                BaseController.showMessage("Новые пароли не совпадают");
+                return;
+            }
+
+            const dynamicKey = "password_repeat";
+            let {[dynamicKey]: _, ...rest} = resObj.object;
+            let dataToSend = {...rest} as ChangePasswordFormModel;
+            console.log('can send data to ChangePassword');
+            console.log(dataToSend);
+            const ctrl = new UserController();
+            await ctrl.changePassword(dataToSend);
+        } else {
+            console.log('can not send Data to Signup Api');
+        }
+
+    }
+
+    async processEditProfileSubmit(e: SubmitEvent) {
+        e.preventDefault();
+        console.log('submit: UserController processEditProfileSubmit');
+        const resObj = validationTypedSubmitHandler<EditProfileFormModel>(e, EditProfileFormModel);
+        console.log(resObj);
+        const canSendDataToLogin = resObj.isValidated;
+        if (canSendDataToLogin) {
+            let dataToSend = resObj.object;
+            console.log('can send data to EditProfile');
+            console.log(dataToSend);
+            const ctrl = new UserController();
+            await ctrl.editProfile(dataToSend);
+        } else {
+            console.log('can not send Data to Signup Api');
+        }
+
+    }
+
+    async processUploadAvatarChange(e: Event) {
+        e.preventDefault();
+        console.log('_avatarInput change');
+        console.log(e);
+
+        const input = e.target as HTMLInputElement;
+        const files = input.files;
+        console.log('files', files);
+
+        if (!files || !!files && files.length === 0) {
```

Comment:
Довольно странное условие, думаю можно его сделать более читаемым

Reply:
Согласен, если есть другое видение как. 
Вообще, это - стандартное давно используемое условие: в данном случае - проверки на отсутствие значения (д.б. массив), или что значение (объект массив) есть, но пустой - нулевой длины.

Т.е. отсутствие фактических данных в пришедшей переменной.

Reply:
Например так явно будет понятно что тут проверяется
```suggestion
        if (Array.isArray(files); && files.length === 0) {
```